src/main/java/core/basesyntax/service/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Inventory;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String getReport(Inventory inventory) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("fruit,quantity\n");
+        for (Map.Entry<String, Integer> entry : inventory.getInventory().entrySet()) {
```

Comment:
Directly accessing the inventory data structure from the `Inventory` class violates encapsulation. Consider adding a method in the `Inventory` class that returns a copy of the data or the necessary data for generating the report, rather than exposing the internal data structure.